### PR TITLE
rebar3Relx: ignore deps in other profiles

### PR DESCRIPTION
**Pull request by dlesl**
###### Motivation for this change

This change switches from modifying `rebar.{config,lock}` to prevent dependencies from being fetched, to modifying rebar3 to respect the env var `REBAR_IGNORE_DEPS`. This is more powerful, since it allows us to ignore dependencies in other profiles (such as test). The usefulness of this is demonstrated by adding `checkPhase` to rebar3 and erlang-ls.

In #119452 we decided not to include a rebar3 plugin directly in the `nixpkgs` repo, however in this case I feel it is the right thing to do because it's more of a patch than a plugin; functionality-wise it's very similar to the existing `skip-plugins.patch`. @gleber do you have any thoughts on this?

The rebar-deps.nix files in this PR were generated using erlang-nix/rebar3_nix#5, I had to make this change to get `proper` to build, and I think it is a generally useful change. @gleber if you agree with those changes should we release a new version and update `rebar3_nix` in nixpkgs?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125965` at 5948bdba run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>elixir_ls</li>
    <li>erlang-ls</li>
    <li>lfe (lfe_1_3)</li>
    <li>plausible</li>
    <li>rebar3</li>
  </ul>
</details>
<details>
  <summary>9 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `runCommand`.
Near pkgs/development/tools/build-managers/rebar3/default.nix:2:34:
```
  |
2 |   fetchHex, erlang, makeWrapper, runCommand,
  |                                  ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/name-and-version.md">name-and-version</a>

Did you mean to pass `pname` instead of `name` to `mkDerivation`?

Near pkgs/development/beam-modules/rebar3-release.nix:44:5:
```
   |
44 |     name = "${name}-${version}";
   |     ^
```

Near pkgs/development/beam-modules/rebar3-release.nix:45:12:
```
   |
45 |     inherit version;
   |            ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/development/beam-modules/erlang-ls/default.nix:29:3:
```
   |
29 |   checkPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/development/tools/build-managers/rebar3/default.nix:42:5:
```
   |
42 |     checkPhase = ''
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near lib/attrsets.nix:212:41:
```
    |
212 |   nameValuePair = name: value: { inherit name value; };
    |                                         ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/development/tools/build-managers/rebar3/default.nix:38:5:
```
   |
38 |     buildPhase = ''
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/beam-modules/rebar3-release.nix:91:5:
```
   |
91 |     meta = {
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/build-managers/rebar3/default.nix:48:5:
```
   |
48 |     installPhase = ''
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near lib/attrsets.nix:212:41:
```
    |
212 |   nameValuePair = name: value: { inherit name value; };
    |                                         ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/build-managers/rebar3/default.nix [hunk 1]
```diff
@@ -1,5 +1,5 @@
 { lib, stdenv, fetchFromGitHub,
-  fetchHex, erlang, makeWrapper,
+  fetchHex, erlang, makeWrapper, runCommand,
```
Comment (happysalada) [inline pkgs/development/tools/build-managers/rebar3/default.nix:2]: are you still using this?

Comment (dlesl) [reply pkgs/development/tools/build-managers/rebar3/default.nix:2]: thanks, missed this!

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 125965` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>plausible</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>elixir_ls</li>
    <li>erlang-ls</li>
    <li>lfe (lfe_1_3)</li>
    <li>rebar3</li>
  </ul>
</details>

**Comment by happysalada** [comment]
The failure is unrelated and is my fault

**Pull request merged**